### PR TITLE
[Issue-3666] Add PWA to docusaurus

### LINKED ISSUE #3666: Add PWA to docusaurus
Our current docs (docusaurus) doesn't support [PWA](https://web.dev/progressive-web-apps/)

There is a pluging in docusaurus where you can enable PWA [plugin-pwa](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-pwa)

### PR DESCRIPTION
**Pull request by Kevan-Y**
## Issue This PR Addresses
Fixes #3666

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
- Add PWA to docusaurus

## Steps to test the PR

- Checkout branch
- Run `pnpm i`
- Run `pnpm --filter @senecacdot/telescope-docs build && pnpm --filter @senecacdot/telescope-docs serve`

Should able to see this button to use PWA
image

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/web/docusaurus/static/manifest.json
Comment (RC-Lee) [inline src/web/docusaurus/static/manifest.json:5]: Do we want to make use of the theme colours defined here
https://github.com/Seneca-CDOT/telescope/blob/232e0a98a852b16335bf610a9e0620311f95ab96/src/web/docusaurus/src/css/custom.css#L10
https://github.com/Seneca-CDOT/telescope/blob/232e0a98a852b16335bf610a9e0620311f95ab96/src/web/docusaurus/src/css/custom.css#L28

**Comment by humphd** [comment]
TIL about [`pnpm --filter`](https://pnpm.io/filtering).  Question: could we use this to improve what we do in `pnpm dev` [here](https://github.com/Seneca-CDOT/telescope/blob/master/package.json#L14)?  cc @cindyledev

**Comment by cindyledev** [comment]
> TIL about [`pnpm --filter`](https://pnpm.io/filtering). Question: could we use this to improve what we do in `pnpm dev` [here](https://github.com/Seneca-CDOT/telescope/blob/master/package.json#L14)? cc @cindyledev

Wow, I didn't know `pnpm --filter` was a thing! It works for `pnpm dev`! I'll create a PR. Thank you based @Kevan-Y

File: src/web/docusaurus/static/manifest.json
Comment (cindyledev) [reply src/web/docusaurus/static/manifest.json:5]: I think we can address this in another issue. I've tried setting `"theme_color": "var(--ifm-color-primary)"` but Chrome dev tools said it was invalid and automatically set the colour to white and I don't want to make this PR bigger than it already is.

**Review by cindyledev — APPROVED** [review]
Works on my machine. Very cool.

File: src/web/docusaurus/static/manifest.json
Comment (RC-Lee) [reply src/web/docusaurus/static/manifest.json:5]: I think it might be
```js
"theme_color": var(--ifm-color-primary)
```
without the quotes around `var()`
but, I am quite weak at front-end so I'll let the front-end gurus decide what to do

Comment (cindyledev) [reply src/web/docusaurus/static/manifest.json:5]: That's what I initially thought, too, but apparently 
```
"theme_color": var(--ifm-color-primary)
``` 
is not valid JSON... this can totally be someone else's issue ;D

Comment (Kevan-Y) [reply src/web/docusaurus/static/manifest.json:5]: I don't think you can use `var` see https://web.dev/learn/pwa/web-app-manifest/#recommended-fields

**Comment by RC-Lee** [comment]
Running into this
```
PS C:\OSD\telescope> pnpm --filter @senecacdot/telescope-docs build

> @senecacdot/telescope-docs@0.0.1 build C:\OSD\telescope\src\web\docusaurus
> docusaurus build

[ERROR] Minimum Node.js version not met :(
[INFO] You are using Node.js v16.13.0, Requirement: Node.js >=16.14.
C:\OSD\telescope\src\web\docusaurus:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @senecacdot/telescope-docs@0.0.1 build: `docusaurus build`
Exit status 1
```

**Comment by Kevan-Y** [comment]
> Running into this
> 
> ```
> PS C:\OSD\telescope> pnpm --filter @senecacdot/telescope-docs build
> 
> > @senecacdot/telescope-docs@0.0.1 build C:\OSD\telescope\src\web\docusaurus
> > docusaurus build
> 
> [ERROR] Minimum Node.js version not met :(
> [INFO] You are using Node.js v16.13.0, Requirement: Node.js >=16.14.
> C:\OSD\telescope\src\web\docusaurus:
>  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @senecacdot/telescope-docs@0.0.1 build: `docusaurus build`
> Exit status 1
> ```

What is your node version?

**Comment by RC-Lee** [comment]
> > Running into this
> > ```
> > PS C:\OSD\telescope> pnpm --filter @senecacdot/telescope-docs build
> > 
> > > @senecacdot/telescope-docs@0.0.1 build C:\OSD\telescope\src\web\docusaurus
> > > docusaurus build
> > 
> > [ERROR] Minimum Node.js version not met :(
> > [INFO] You are using Node.js v16.13.0, Requirement: Node.js >=16.14.
> > C:\OSD\telescope\src\web\docusaurus:
> >  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @senecacdot/telescope-docs@0.0.1 build: `docusaurus build`
> > Exit status 1
> > ```
> 
> What is your node version?

Right now I'm trying to build the containers in docker to see if it works. 
But think its worth noting to see how it might clash with our node version in `package.json`
https://github.com/Seneca-CDOT/telescope/blob/3f7f5e4bfd0baadcc455186b7e26d30ed353a621/package.json#L86

**Comment by RC-Lee** [comment]
I'm a little confused with PWA and what it does. 
Is there a place I'm supposed to toggle it on? 
image

**Comment by Kevan-Y** [comment]
> Is there a place I'm supposed to toggle it on?

It suppose to have but running on dev mode it won't be shown

**Comment by RC-Lee** [comment]
> > Is there a place I'm supposed to toggle it on?
> 
> It suppose to have but running on dev mode it won't be shown

You might want some more experienced (with docusaurus and front-end) reviewers to review this, since this is out of my comfort zone. 
I can go the easy way and update my node version, but once again I don't know how that might conflict with our `package.json`
https://github.com/Seneca-CDOT/telescope/blob/3f7f5e4bfd0baadcc455186b7e26d30ed353a621/package.json#L86

And since its hidden in dev mode, I can't see if its working through docker containers locally

**Comment by Kevan-Y** [comment]
@RC-Lee I believe your browser is running under Firefox.
>Note: Currently, only the Chromium-based browsers such as Chrome, Edge, and Samsung Internet require the service worker. If developing your app using Firefox, be aware that you will need a service worker to be compatible with Chromium-based browsers.
see https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Installable_PWAs#requirements

**Comment by Kevan-Y** [comment]
@humphd @cindyledev @RC-Lee  Wondering if it would be good to have a button to install app, so it would be more accessible for people that doesn't know what is PWA.

image

**Comment by humphd** [comment]
Having a link to do it makes sense I think, yes.

**Comment by Kevan-Y** [comment]
@humphd Added

**Review by humphd — APPROVED** [review]

**Comment by Kevan-Y** [comment]
@cindyledev Just need a re-approval, then I will squash those commit and merge

**Comment by humphd** [comment]
Why are we merging master into this branch?

Screen Shot 2022-10-12 at 9 44 20 AM

**Comment by Kevan-Y** [comment]
> Why are we merging master into this branch?
> 
> Screen Shot 2022-10-12 at 9 44 20 AM

So I could use Squash and the merge button from GitHub. 
image
I needed to make sure my branch have the most updated master commit.

**Comment by Kevan-Y** [comment]
> I don't care how you want to do it so long as we don't end-up with `master` merged into topic branches. Ideally you'd squash on your own and merge that.

Alright, I rebased/Squash manually

**Review by cindyledev — APPROVED** [review]
Pretty slick, if I do say so myself

2022-10-12 13_01_30-Telescope Documentation - Environment Setup _ Telescope

**Pull request merged**
